internal/agent/delta/last_entityId_test.go

```diff
@@ -0,0 +1,207 @@
+
+	le := &EntityIDFilePersist{
+		readFile: func(path string) (entity.ID, error) {
+			return entity.EmptyID, fmt.Errorf("should not read from file")
+		},
+		lastEntityID: expectedID,
+	}
+
+	id, err := le.GetEntityID()
+
+	assert.Equal(t, expectedID, id)
+	require.NoError(t, err)
+}
+
+func TestEntityIDFilePersist_ErrWhenReadingFile(t *testing.T) {
+	expectedMessage := "failed when reading file"
+
+	le := &EntityIDFilePersist{
+		readFile: func(path string) (entity.ID, error) {
+			return entity.EmptyID, fmt.Errorf(expectedMessage)
+		},
+	}
+
+	id, err := le.GetEntityID()
+
+	assert.Equal(t, id, entity.EmptyID)
+	assert.Error(t, err, expectedMessage)
+}
+
+func TestEntityIDFilePersist_UpdateValue(t *testing.T) {
+	expectedID := entity.ID(10)
+
+	le := &EntityIDFilePersist{
+		writeFile: func(id entity.ID, filePath string) error {
+			return nil
+		},
+	}
+
+	err := le.UpdateEntityID(expectedID)
+	require.NoError(t, err)
+	assert.Equal(t, expectedID, le.lastEntityID)
+}
+
+func TestEntityIDFilePersist_ErrWhenWritingFile(t *testing.T) {
+	expectedErrMessage := "file could not be written"
+	expectedID := entity.ID(10)
+
+	le := &EntityIDFilePersist{
+		writeFile: func(id entity.ID, filePath string) error {
+			return fmt.Errorf(expectedErrMessage)
+		},
+	}
+
+	err := le.UpdateEntityID(expectedID)
+
+	assert.Errorf(t, err, "Update lastEntityID should return an error when failed writing file")
+	assert.Equal(t, expectedErrMessage, err.Error())
+	assert.Equal(t, expectedID, le.lastEntityID)
+}
+
+// Read File IT
+func TestReadFile_ReturnContent(t *testing.T) {
+	expected := entity.ID(10)
+
+	path, err := TempDeltaStoreDir()
+	assert.NoError(t, err)
+
+	file := filepath.Join(path, "temporary_file")
+	err = ioutil.WriteFile(file, []byte(expected.String()), 0644)
+	assert.NoError(t, err)
+
+	content, err := readFileFn(file)
+
+	assert.Equal(t, expected, content)
+	assert.NoError(t, err)
+}
+
+func TestReadFile_EmptyFile(t *testing.T) {
+	path, err := TempDeltaStoreDir()
+	assert.NoError(t, err)
+
+	file := filepath.Join(path, "temporary_file")
+	err = ioutil.WriteFile(file, []byte(""), 0644)
+	assert.NoError(t, err)
+
+	content, err := readFileFn(file)
+
+	assert.Equal(t, entity.EmptyID, content)
+	require.Error(t, err, "Expected to return an Empty file error")
+}
+
+func TestReadFile_FileNotFound(t *testing.T) {
+	content, err := readFileFn("some_non-existing_file_path")
+
+	assert.Equal(t, entity.EmptyID, content)
+	require.NoError(t, err)
+}
+
+func TestReadFile_NoPermission(t *testing.T) {
+	path, err := TempDeltaStoreDir()
+	assert.NoError(t, err)
+
+	file := filepath.Join(path, "temporary_file")
+	err = ioutil.WriteFile(file, []byte(""), 0000)
+	assert.NoError(t, err)
+
+	content, err := readFileFn(file)
+
+	assert.Equal(t, entity.EmptyID, content)
+	require.Error(t, err, "Expected to return an permission error")
+}
+
+func TestReadFile_ErrParseContent(t *testing.T) {
+	temp, err := TempDeltaStoreDir()
+	require.NoError(t, err)
+
```

Comment:
```suggestion
	defer func() {
		_ = os.RemoveAll(temp)
	}()
```
Always good to clean up after running tests.

Reply:
Thanks, I will do it for all tests.